### PR TITLE
unwatch object not closing the websocket

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1369385

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett PTAL

File: app/scripts/services/data.js [hunk 1]
```diff
@@ -559,7 +559,7 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
     var existingWatchOpts = this._watchOptions(key);
     if (existingWatchOpts) {
       // Check any options for compatibility with existing watch
-      if (existingWatchOpts.poll != opts.poll) {
+      if (!!existingWatchOpts.poll != !!opts.poll) { // jshint ignore:line
```
Comment (spadgett) [inline app/scripts/services/data.js]: It could be a `!==` now that we are coercing to a boolean.

**Comment by spadgett** [comment]
One nit. LGTM

**Comment by jwforres** [comment]
[merge]

File: app/scripts/services/data.js [hunk 2]
```diff
@@ -559,7 +559,7 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
     var existingWatchOpts = this._watchOptions(key);
     if (existingWatchOpts) {
       // Check any options for compatibility with existing watch
-      if (existingWatchOpts.poll != opts.poll) {
+      if (!!existingWatchOpts.poll !== !!opts.poll) { // jshint ignore:line
```
Comment (spadgett) [inline app/scripts/services/data.js]: Do we still need `ignore:line` with `!==` ?

**Pull request merged**